.github/workflows/actions/ubuntu-build-deps/action.yml

```diff
@@ -0,0 +1,50 @@
+name: 'ubuntu build dependencies'
+description: 'install required build dependencies for ubuntu'
+inputs:
+
+  SUDO:
+    description: "set to true to run apt as root"
+    required: false
+    default: ""
+
+
+runs:
+  using: 'composite'
+  steps:
+    - name: apt update
+      shell: bash
+      run: ${{inputs.SUDO}} apt update
+
+    - name: install build tools
+      shell: bash
+      run: |
+        ${{inputs.SUDO}} apt install -y \
+            build-essential \
+            clang \
+            clang-tidy \
+            ninja-build \
+            iwyu \
+            pkgconf \
```

Comment:
clang, clang-tidy, ninja-build, and iwyu are only used by the cmake and clang-tidy CI workflows, as far as I know.

Is it worth it to keep these for all of the other potential builds types?

Reply:
they could very easily be extracted to a different action, to be called by cmake and clang-tidy if necessary